### ISSUE TITLE
optimizeCss v0.11.0 breaks image styling on production builds with Sveltekit

### ISSUE THREAD

**Issue by andrewrisse** [carbon-design-system/carbon-preprocess-svelte#55]
### Environment
Device and OS: Macbook Pro 16, Sonoma 14.3.1, Google Chrome Version 123.0.6312.86 (Official Build) (arm64)
App/package versions: 
carbon-preprocess-svelte: v0.11.0
carbon-components-svelte: 0.85.0
svelte: 4.2.12

### Steps to reproduce
1. Import optimizeCSS in accordance with [docs](https://github.com/carbon-design-system/carbon-preprocess-svelte#sveltekit)
2. Run a production build: npm run build
3. View application in browser (recommend clear cache and hard reload)
4. Images no longer have styling applied to them and are full screen.

Example code affected:  

```
	<div class="logo">
		<img alt="Logo" src={logo} class="logo" />
	</div>

<style lang="scss">
	.logo {
		width: 252px;
		height: 72px;
	}
</style>
```

I confirmed that removing the optimizeCSS plugin and rebuilding fixes the images and they display as expected.

I also noticed some other components other than images were affected. We have SideNavMenuItems with an OverflowMenu in them that was no longer styled correctly. This component has a lot of overrides of default Carbon classes, but the optimizeCSS plugin seems to have affected those overrides.

Running locally in development mode (npm run dev), images are not affected.

### Screenshots of side nav items
Without optimizeCSS:

<img width="236" alt="Screenshot 2024-03-27 at 9 51 08 AM" src="https://github.com/carbon-design-system/carbon-preprocess-svelte/assets/52644157/32bf672c-4a4b-4064-8286-a6009c9f4b58">

With optimizeCSS:

<img width="243" alt="Screenshot 2024-03-27 at 9 50 20 AM" src="https://github.com/carbon-design-system/carbon-preprocess-svelte/assets/52644157/2b8001a5-aeb9-417e-be88-bac76b70016a">

**Comment by Kuchteq** [comment]
Facing the same issue with basically any custom theming inside <style> tags, whether it being locally scoped or global. Maybe adding bx-- would resolve this but I wouldn't propose it as a solution due to possible name conflicts

**Comment by metonym** [comment]
I'm very sorry you ran into this.

I've published a patch in [v0.11.1](https://github.com/carbon-design-system/carbon-preprocess-svelte/releases/tag/v0.11.1) that fixes this. It uses an even more conservative approach to first check that a selector contains the Carbon `bx--` prefix.

In the screenshot below, I've tested the following:

- Selector from scoped Svelte style is preserved
- Selector from global Svelte style (`:global()`) is preserved
- Selector from imported CSS StyleSheet is preserved
- Selector overriding `.bx--` selector is preserved

<img width="1287" alt="Screenshot 2024-04-07 at 8 09 23 PM" src="https://github.com/carbon-design-system/carbon-preprocess-svelte/assets/10718366/e09b5191-018c-4eb6-96fc-53ef4a08edbc">

**Comment by metonym** [comment]
To further walk the walk, I've [applied `optimizeCss`](https://github.com/carbon-design-system/sveld/pull/124) to the `sveld` playground.

I've verified that a modest amount of unused Carbon selectors are removed, without modifying third-party styles (e.g., CodeMirror).

```sh
Optimized assets/index-Dy43zAwI.css
Before: 750.82 kB
After:  336.35 kB (-55.2%)

Optimized assets/CodeHighlighter-1BpQ5oIG.css
Before: 2.3 kB
After:  2.3 kB (-0%)
```
